src/functions/gradients.jl

```diff
@@ -1,3 +1,180 @@
+@doc raw"""
+    ∇acceleration_bezier(
+        M::Manifold,
+        B::AbstractVector{P},
+        degrees::AbstractVector{<:Integer}
+        T::AbstractVector{<:AbstractFloat}
+    )
+
+compute the gradient of the discretized acceleration of a (composite) Bézier curve $c_B(t)$
+on the `Manifold` `M` with respect to its control points `B` given as a point on the
+`PowerManifold` assuming C1 conditions and known `degrees`. The curve is
+evaluated at the points given in `T` (elementwise in $[0,N]$, where $N$ is the
+number of segments of the Bézier curve). The [`get_bezier_junctions`](@ref) are fixed for
+this gradient (interpolation constraint). For the unconstrained gradient,
+see [`∇L2_acceleration_bezier`](@ref) and set $λ=0$ therein. This gradient is computed using
+[`adjoint_Jacobi_field`](@ref)s. For details, see [^BergmannGousenbourger2018].
+See [`de_casteljau`](@ref) for more details on the curve.
+
+# See also
+
+[`cost_acceleration_bezier`](@ref),  [`∇L2_acceleration_bezier`](@ref), [`cost_L2_acceleration_bezier`](@ref).
+
+[^BergmannGousenbourger2018]:
+    > Bergmann, R. and Gousenbourger, P.-Y.: A variational model for data fitting on
+    > manifolds by minimizing the acceleration of a Bézier curve.
+    > Frontiers in Applied Mathematics and Statistics (2018).
+    > doi [10.3389/fams.2018.00059](http://dx.doi.org/10.3389/fams.2018.00059),
+    > arXiv: [1807.10090](https://arxiv.org/abs/1807.10090)
+"""
+function ∇acceleration_bezier(
+    M::Manifold,
+    B::AbstractVector{P},
+    degrees::AbstractVector{<:Integer},
+    T::AbstractVector{<:AbstractFloat},
+) where {P}
+    gradB = _∇acceleration_bezier(M, B, degrees, T)
+    Bt = get_bezier_segments(M, B, degrees, :differentiable)
+    for k=1:length(Bt) # we interpolate so we do not move end points
+        gradB[k].pts[end] .= zero_tangent_vector(M, Bt[k].pts[end])
```

Comment:
I think you could use `zero_tangent_vector!` instead of broadcasting.

Reply:
Good idea, thanks!